rasa/utils/tensorflow/layers.py

```diff
@@ -589,8 +591,15 @@ def __init__(
         self.use_max_sim_neg = use_max_sim_neg
         self.neg_lambda = neg_lambda
         self.scale_loss = scale_loss
-        self.parallel_iterations = parallel_iterations
         self.same_sampling = same_sampling
+        self.constrain_similarities = constrain_similarities
+        self.model_confidence = model_confidence
+        self.similarity_type = similarity_type
+        if self.similarity_type and self.similarity_type not in {COSINE, INNER}:
```

Comment:
what happens `if not self.similarity_type`?

Reply:
Good point, used it as it was before. Changed it now to `if not self.similarity_type or self.similarity_type not in {COSINE, INNER}`

Reply:
Good point, used it as it was before. Changed it now to `if not self.similarity_type or self.similarity_type not in {COSINE, INNER}`

Reply:
I suggest to remove `Optional` from input type and don't check for `None`. We cannot check every argument